### PR TITLE
Add methods that parse a URI

### PR DESCRIPTION
**Pull request by erikcorry**
Instead of separate host and path args, you can now specify the whole URI in one string.

Also fixes bugs where redirects between HTTP and HTTPS did not work.

We would like to deprecate giving the host in the form "hostname:portnumber" so that you cleanly choose between a parsed input and an itemized input.  Unfortunately there's no such thing as a runtime deprecation.

No Punycode support, so non-ASCII host name must be pre-expanded to the xn--foo-bar notation.  It's unclear to me whether redirects are expected to handle that, or whether the Location: header in the redirect has
already puny-endoded host names.

### CODE REVIEW

File: src/client.toit [hunk 1]
```diff
@@ -71,33 +71,66 @@ class Client:
...
 
-  use_tls_ ::= false
+  use_tls_by_default_ ::= false
   certificate_/tls.Certificate? ::= null
   server_name_/string? ::= null
   root_certificates_/List ::= []
 
   /**
   Constructs a new client instance over the given interface.
+  The client will default to a insecure HTTP connection, but this can be
```
Comment (kasperl) [inline src/client.toit:81]:
```suggestion
  The client will default to an insecure HTTP connection, but this can be
```

File: src/client.toit [hunk 2]
```diff
@@ -153,35 +187,64 @@ class Client:
...
+
+  /**
+  Makes an HTTP connection, then immediately upgrades to a $WebSocket connection.
+  Connects to the given URI, of the form "wss://api.example.com:1080/path/to/end-point".
+  On an error, the block is called with the $Response as its argument.
+  In the case that there are too many redirects, the block is called with null
+    as its argument.
+  After this call, this client can no longer be used for regular HTTP requests.
+  */
+  web_socket --uri/string --headers=Headers --follow_redirects/bool=true [on_error] -> WebSocket:
```
Comment (floitsch) [inline src/client.toit:240]: Not sure the `[on_error]` functionality is necessary, but no need to change.

File: src/client.toit [hunk 3]
```diff
@@ -255,12 +324,35 @@ class Client:
...
+    parsed := parse_ uri host port path use_tls
+
+    return post_ data parsed --headers=headers --content_type=content_type --follow_redirects=follow_redirects
+
+  parse_ uri/string? host/string? port/int? path/string? use_tls/bool? -> ParsedUri_:
+    if uri:
+      if host or port or path or use_tls: throw "Cannot combine --uri with host, port or path arguments"
+      return ParsedUri_.parse uri
+    else:
+      if not host or not path: throw "Must specify either --uri or --host and --path"
```
Comment (floitsch) [inline src/client.toit:345]:
```suggestion
      if not host or not path: throw "Must specify either --uri, or --host and --path"
```

**Review by floitsch — APPROVED** [review]
LGTM.

**Pull request merged**